performance-harness/proposals/performance_testing_harness_stage_3.md

```diff
@@ -0,0 +1,107 @@
+# Performance Testing Harness Stage 3 Planning
+
+Roadmap for the development of a **Performance Testing Harness** -- Stage 3.
+
+Requirements are being curated for the development of a performance testing and evaluation harness here: [Performance Testing Harness Requirements & Planning & Staging](https://github.com/eosnetworkfoundation/product/tree/main/performance-harness/proposals)
+
+
+## Stage 3
+
+###   Multi-producer & "relay" or "canary" node configuration support (Carried over from Stage 2)
+    -   Add configurable network topology based on public networks
+        -   multiple producer nodes, each accessed through a dedicated relay/canary node
+        -   all relay nodes connected in mesh
+    -   Include configurable "relay" nodes
+        -   Specific configuration per node type
+    -   Expand reporting to include multiple nodes
+
+### Work with ENF to identify additional metrics that would be useful to publish (Carried over from Stage 2)
+    -   Support additional metrics that would be useful to publish
+
+### HTTP harness testing
+
+-   Setup HTTP performance testing across single node
+    -   Transaction generator to be able to target the http interface
+        -   Need new http transaction provider using asynchronous requests
+    -   Support http_plugin on canary node
+    -   Performance Test Harness to be able to select which protocol to use (http vs. p2p)
+    -   Support Transfer Per Second transaction generation
+    -   Support JSON transaction generation
+
+### Expand Transaction generator test scenarios
+
+-   EVM Transactions (Carried over from Stage 2)
+-   Large action
+    -   Large action payload (~200KB string payload) that is not used
+-   DB intense actions
+    -   Lots of reads with no writes
+    -   Lots of writes with no reads
+    -   Mix of reads/writes
+-   Kitchen Sink (mix of all transaction types)?
+
+### Additional Latency Measuremeants
+
+-   Measure latency:
+    -   From creation
+    -   From consumption in nodeos
+    -   From consumption from SHiP
+    -   From consumption from trace_api_plugin
+
+### Create Replay Performance Test

```

Comment:
This will overlap with another internal ENF project that allows us to do a distributed replay of a full blockchain to guarantee new code doesn't alter consensus level history. 

Internally we call this the "chicken test" for reasons I don't remember but may have to do with Legend of Zelda. 

We should remove this from the Performance Harness track and instead create a new scope that extracts measurements from the "chicken test" when it is online.  There may need to be extensions to that test rigging if we want to measure throughput rather than integrity but, I suspect we will want to maximize throughput in both cases.